modules/core/src/main/scala/sangria/validation/rules/OverlappingFieldsCanBeMergedFast.scala

```diff
@@ -0,0 +1,70 @@
+package sangria.validation.rules
+
+import java.util.function.Consumer
+
+import sangria.ast
+import sangria.ast.AstVisitorCommand
+import sangria.validation._
+import sangria.validation.rules.overlappingfields.{CachedCheck, SelectionBuilder, SelectionConflictViolationsBuilder, SelectionContainer}
+
+/**
+  * Overlapping fields can be merged
+  *
+  * A selection set is only valid if all fields (including spreading any
+  * fragments) either correspond to distinct response names or can be merged
+  * without ambiguity.
+  *
+  * The algorithm is described in [[CachedCheck]].
+  */
+class OverlappingFieldsCanBeMergedFast extends ValidationRule {
```

Comment:
What is your opinion on adding an annotation like akka is doing with`@ApiMayChange` so that we can rename it in the future?

Reply:
I would be fine with that. Is there already an established convention how to signify that in sangria?

Ultimately I hope this can replace the previous rule, so the option to choose explicitly could also disappear at some point.

Reply:
Thinking about this, I would probably prefer using packages for this type of information, as an annotation can go unnoticed by the client, while a package like "internal" or "experimental" has to be referenced in the import by a client. So anyone importing something from an "experimental" package states that he knows what he is doing.

Reply:
> I would be fine with that. Is there already an established convention how to signify that in sangria?

I don't think there is an existing convention.
I'd recommend to "copy" what others are doing, like in the akka project, unless someone has a better idea.

> Ultimately I hope this can replace the previous rule, so the option to choose explicitly could also disappear at some point.

👍

Reply:
What would be the expectation when using `@ApiMayChange`? Does it still obey semantic versioning?

Reply:
@yann I'd leave the decision here to you and the other maintainers. Maybe there is a way that we could circumvent the problem and come up with a better name? `@ApiMayChange` is something I am not particularly fond of, due to how it allows invisible exceptions to semantic versioning. A separately versioned `contrib` module from which code then is migrated to `core` could be a clearer solution. But to be honest I'd prefer to get this rule in without that overhead.

Reply:
> I'd leave the decision here to you and the other maintainers. Maybe there is a way that we could circumvent the problem and come up with a better name? `@ApiMayChange` is something I am not particularly fond of, due to how it allows invisible exceptions to semantic versioning. A separately versioned `contrib` module from which code then is migrated to `core` could be a clearer solution. But to be honest I'd prefer to get this rule in without that overhead.

I'm not the "maintainer", just giving some of my time to help.
I also think a separately versioned `contrib` module would be cleaner.
But giving the time we all can give, I'd suggest to go with the `@ApiMayChange` annotation for now so that we can merge this.